pkg/controller/job/job_controller.go

```diff
@@ -646,6 +650,28 @@ func pastBackoffLimitOnFailure(job *batch.Job, pods []*v1.Pod) bool {
 	return result >= *job.Spec.BackoffLimit
 }
 
+// pastZeroBackoffLimitOnFailure checks if container restartCounts sum exceeds backoffLimit.
+// This method applies only to job with backoffLimit == 0 and restartPolicy == OnFailure.
+// If any of the container has failed once, the job passes backoffLimit.
+func pastZeroBackoffLimitOnFailure(pods []*v1.Pod) bool {
```

Comment:
Why we need another function which is identical to `pastBackoffLimitOnFailure` ? Why the calculations from there can't be used in here?